challenges/type-unions/prompt.md

```diff
@@ -0,0 +1,70 @@
+## What Problem Type Unions Solve
+
+Type In a previous example for `type-aliases` we saw a problem.  We were writing code for a space shuttle and we wanted to make sure that we can't accidentally use the wrong units type.
+
+```ts
+type Meters = number;
+type Miles = number;
+
+const landSpacecraft = (distance: Meters) {
+  // ... do fancy math ...
+}
+```
+
+But we noticed when we call `landSpacecraft` with the wrong literal type (see [the challenge on literal types](todo-link)), there's no error like we might expect:
+
+```ts
+const distanceInMiles: Miles = 1242;
+landSpacecraft(distanceInMiles);
+```
+
+What do we do?  We have some places in our code that take a unit of `miles` and some places in our code that take a unit of `meters` and we want to be sure we _NEVER_ mix the two up since it could mean literally destroying the entire spacecraft if we do (even once).
+
+We can use unions!
+
+## How To Use Type Unions
+
+Unions are defined using a single "pipe" character: `|` separating two types or type aliases.
+
+Let's take a look:
+
+```ts
+type Meters = number;
+type Miles = number;
+
+type Distance = Meters | Miles;
+```
+
+That's a good start, but there's still no way for our function to tell the difference between the two, so we _spruce up_ the first two type aliases a bit.
+
+```ts
+type Meters = {
+  unit: 'meters';
+  value: number;
+};
+
+type Miles = {
```

Comment:
should we even go simpler and have discriminated unions as a separate challenge?

Reply:
I'd love that (and, indeed, actually started that way) but the problem I had was demonstrating _why it's useful_ without some kind of concrete example.  I guess that doesn't have to happen in _every single challenge_ but it's a goal of mine to try to keep them practical.

any ideas on other examples to use?